### AVI SSVEP

#### Carregamento da base de dados

Classificação por isolamento de frequência.

Objetivos:

- Padronizar os dados - multi-target - dos participantes.

Nesse arquivo estaremos realizando o carregamento das informações das bases de dados do AVI que estão distribuídos em diversos arquivos, e realizaremos uma pequena formatação para facilitar o preprocessamento dos dados na fase seguinte.

In [1]:
from scipy.io import loadmat
import numpy as np

In [2]:
avi_data = list()
for i in range(1,6):
    for j in range(1,3):
        avi_data.append(loadmat(f"../datasets/avi/multi/Sub{i}_{j}_multitarget.mat"))

np.array(avi_data).shape


(10,)

#### Carregamento dos dados

Para realizar o carregamento dos dados obtidos a partir da base de dados, precisaremos separar as informações em 'dados'(data) e 'labels' para relacionar as informações com as frequencias utilizadas.

as informações dos dados estçao dentro do índice `['Data'][0][0][7]` do nosso array avi_data.
ja as informações dos dados das labels (frequencias) estão no índice `['Data'][0][0][8][0]`

In [3]:
X = [d['Data'][0][0][7] for d in avi_data]
y = [d['Data'][0][0][8][0] for d in avi_data]

print(np.array(X).shape)

print("Divisão dos dados:")
for data in X:
    print(data.shape)

print("\nDivisão dos rótulos (sequências):")
for data in y:
    print(data)

(10, 8192, 10)
Divisão dos dados:
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)

Divisão dos rótulos (sequências):
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]


#### Formatação dos dados

O padrão original dos dados vem no formato `(10, 8192, 10)`, que representam 10 registros de voluntários, com 8192 dados (instantes que os receptores captam as informações do cérebro quando as frequencias piscam), de 10 frequencias diferentes.

Como existe uma relação clara entre as o voluntário com as frequencias que foram testadas em cada um deles, podemos unir essas colunas sem perder informações relevantes.

In [4]:
X = np.array(X)
print(X.shape)
X = X.swapaxes(1,2)
print(X.shape)
X = X.reshape(X.shape[0] * X.shape[1], X.shape[2])
X.shape


(10, 8192, 10)
(10, 10, 8192)


(100, 8192)

In [6]:
# salva os dados numpy

np.save("../datasets/avi/multi/data.npy", X)
np.save("../datasets/avi/multi/labels.npy", y)